test_rmw_implementation/test/test_publisher.cpp

```diff
@@ -375,3 +375,59 @@ TEST_F(CLASSNAME(TestPublisherUse, RMW_IMPLEMENTATION), count_mismatched_subscri
   EXPECT_EQ(RMW_RET_OK, ret) << rmw_get_error_string().str;
   EXPECT_EQ(0u, subscription_count);
 }
+
+class CLASSNAME (TestPublisherUseLoan, RMW_IMPLEMENTATION)
+  : public CLASSNAME(TestPublisherUse, RMW_IMPLEMENTATION)
+{
+protected:
+  using Base = CLASSNAME(TestPublisherUse, RMW_IMPLEMENTATION);
+
+  void SetUp() override
+  {
+    Base::SetUp();
+    // Check if loaning is supported by the implementation
+    if (!pub->can_loan_messages) {
+      GTEST_SKIP();
+    }
+  }
+
+  void TearDown() override
+  {
+    Base::TearDown();
+  }
+};
+
+TEST_F(CLASSNAME(TestPublisherUseLoan, RMW_IMPLEMENTATION), borrow_loaned_message_bad_arguments) {
+  void * msg_pointer = nullptr;
+  rmw_ret_t ret = rmw_borrow_loaned_message(nullptr, ts, &msg_pointer);
+  EXPECT_EQ(RMW_RET_INVALID_ARGUMENT, ret) << rmw_get_error_string().str;
+  rmw_reset_error();
+
+  EXPECT_EQ(nullptr, msg_pointer);
```

Comment:
@Lobotuerk nit^2: here and below, I'd tuck this expectation next to the previous call which is the one that could've (incorrectly) modified it.

Reply:
Changed